app/models/application_history_item.rb

```diff
@@ -0,0 +1,48 @@
+#
+# View Object for an application history item
+#
+class ApplicationHistoryItem < ApplicationStruct
+  VIEWABLE_DATA = %i[from_whom_name to_whom_name].freeze
+
+  attribute :user_id, Types::Uuid.optional
+  attribute :user_name, Types::String
+  attribute :timestamp, Types::Nominal::DateTime
+  attribute :event_type, Types::String
+  attribute :viewable_data, Types::Hash.default({}.freeze)
+
+  def description
+    I18n.t(
+      event_type.underscore.tr('/', '.'),
+      scope: 'event.description',
+      **viewable_data
+    )
+  end
+
+  class << self
+    def from_event(event)
+      user_id = event.data.fetch(:user_id)
+
+      new(
+        user_id: user_id,
+        user_name: User.name_for(user_id),
+        timestamp: event.timestamp,
+        event_type: event.event_type,
+        viewable_data: viewable_data_for_event(event)
+      )
+    end
+
+    def viewable_data_for_event(event)
+      case event.event_type
```

Comment:
Maybe this could be a hash look up? Might save some space - especially as we don't have a 'else' catch-all

Reply:
One thing we could do is just have this as part of the event class as `data_for_event` then just call it like so:

```ruby
def viewable_data_for_event(event)
  event.data_for_event
end
```

That way we avoid this case statement completely, and we rely on event polymophism to dictate behaviour.

Reply:
Agree. Will flag for a later PR.